src/filesystem/implementation.js

```diff
@@ -1873,7 +1873,7 @@ function writeFile(context, path, data, options, callback) {
   if(typeof data === 'number') {
     data = '' + data;
   }
-  if(typeof data === 'string' && options.encoding === 'utf8') {
+  if(typeof data === 'string' && (options.encoding || 'utf8') === 'utf8') {
```

Comment:
This limits us to only ever allowing `'utf8'`, since this doesn't propagate the encoding into `Buffer.from()` as the second arg, and Buffer/node support other encoding types.  What about something like:

```js
if(!Buffer.isBuffer(data)) {
  if(typeof data !== 'string') {
    data = Buffer.from(data.toString());
  } else {
    data = Buffer.from(data, options.encoding || 'utf8');
  }
}
```

Untested code, just thinking.

Reply:
Yeah, that's better for sure =P I also moved the typeof number check into the !isBuffer if block